mentoring/programs/mentoring-lead.md

```diff
@@ -0,0 +1,40 @@
+# Mentoring Lead Handbook
+
+## Overview
+
+The Mentoring Lead role is created to help upstream Kubernetes contributors to participates in various mentorship programs like [GSoC](mentoring/programs/google-summer-of-code.md), [GSoD](mentoring/programs/google-season-of-docs.md), [LFX Mentorship](mentoring/programs/lfx-mentorship.md) and many more as an independent organisation or under Cloud Native Computing Foundation(CNCF). 
+
+## Responsibilities
```

Comment:
I'd like to see these smoothed out more and called out with some specific actions that we already know need to happen. some of those are on the issue https://github.com/kubernetes/community/issues/6471